src/Menu/MenuFactory.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace CubeAgency\ArboryMenu\Menu;
+
+use Arbory\Base\Nodes\Node;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Request;
+
+class MenuFactory
+{
+    /** @var Menu */
+    protected $menu;
+
+    /**
+     * MenuFactory constructor.
+     * @param Menu $menu
+     */
+    public function __construct(Menu $menu)
+    {
+        $this->menu = $menu;
+    }
+
+    /**
+     * @return Collection
+     */
+    public function build(): Collection
+    {
+        $items = $this->menu->items()->with('relation.related')->get()
+            ->map(function (Item $item) {
+                $link = $item->link;
+                $name = $item->name;
+                if ($item->relation) {
+                    /** @var Node $node */
+                    $node = $item->relation->getAttribute('related');
+                    if ($node !== null && $node instanceof Node) {
+                        $link = $link ?? $node->getUrl('index');
+                        $name = $name ?? $node->name;
+                    }
+                }
+                if (!empty($link) && !empty($name)) {
+                    return [
+                        'link' => $link,
+                        'name' => $name,
+                        'active' => $this->isActive($link)
+                    ];
+                }
+                return false;
+            })
+            ->filter()
+            ->toArray();
```

Comment:
Is this needed? $items variable should already be a collection.

Reply:
This is not necessary. 
Fixed https://github.com/kristapsbworks/arbory-menu/commit/3487eeb40326f812c63c539ade9e59a2eb4dd7d1